handlers/libhoney_event_handler.go

```diff
@@ -0,0 +1,159 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/honeycombio/honeycomb-network-agent/assemblers"
+	"github.com/honeycombio/honeycomb-network-agent/config"
+	"github.com/honeycombio/honeycomb-network-agent/utils"
+	"github.com/honeycombio/libhoney-go"
+	"github.com/rs/zerolog/log"
+
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+)
+
+// libhoneyEventHandler is an event handler that sends events using libhoney
+type libhoneyEventHandler struct {
+	k8sClient  *utils.CachedK8sClient
+	eventsChan chan assemblers.HttpEvent
+}
+
+// NewLibhoneyEventHandler creates a new event handler that sends events using libhoney
+func NewLibhoneyEventHandler(config config.Config, k8sClient *utils.CachedK8sClient, eventsChan chan assemblers.HttpEvent, version string) EventHandler {
+	initLibhoney(config, version)
+	return &libhoneyEventHandler{
+		k8sClient:  k8sClient,
+		eventsChan: eventsChan,
+	}
+}
+
+// Start starts the event handler and begins handling events from the events channel
+// When the context is cancelled, the event handler will stop handling events
+func (handler *libhoneyEventHandler) Start(ctx context.Context) {
+	var event assemblers.HttpEvent
+	for {
+		select {
+		case <-ctx.Done():
+			// close libhoney to ensure all events are sent
+			libhoney.Close()
```

Comment:
We'll need to ensure that this is called to flush pending events to send before process exit. Probably in a follow-up PR because we have a lot of WIP right now.